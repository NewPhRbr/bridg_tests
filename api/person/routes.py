from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import orm

from api.db import get_db
from api.person import schema
from api.util import setattrs
from umdb import person as model

router = APIRouter(prefix="/persons", tags=["persons"])


@router.get("/", response_model=list[schema.Person])
def index(db: orm.Session = Depends(get_db)):
    return db.query(model.Person).all()


@router.post("/", response_model=schema.Person)
def create(body: schema.PersonCreate, db: orm.Session = Depends(get_db)):
    name = setattrs(model.Name(), body.name.model_dump())
    person = setattrs(model.Person(), body.model_dump(exclude={"name"}))
    person.name.append(name)
    db.add(person)
    db.commit()
    db.refresh(person)
    return person


@router.get("/{id}", response_model=schema.Person)
def read(id: int, db: orm.Session = Depends(get_db)):
    person = db.query(model.Person).where(model.Person.id == id).first()
    if person is None:
        raise HTTPException(404)
    return person


@router.patch("/{id}", response_model=schema.Person)
def update(id: int, body: schema.PersonUpdate, db: orm.Session = Depends(get_db)):
    person = db.query(model.Person).where(model.Person.id == id).first()
    if person is None:
        raise HTTPException(404)
    person = setattrs(person, body.model_dump())
    db.add(person)
    db.commit()
    db.refresh(person)
    return person


@router.delete("/{id}")
def delete(id: int, db: orm.Session = Depends(get_db)):
    person = db.query(model.Person).where(model.Person.id == id).first()
    if person is None:
        raise HTTPException(404)
    db.delete(person)
    db.commit()
