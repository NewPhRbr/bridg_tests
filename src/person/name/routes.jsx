import * as personAPI from "../api";
import * as nameAPI from "./api";
import Edit from "./edit";

export default [
  {
    path: ":nameId",
    children: [
      {
        path: "edit",
        Component: Edit,
        loader: async ({ params: { personId, nameId } }) => ({
          person: await personAPI.find(personId).then((x) => x.json()),
          name: await nameAPI.find(personId, nameId).then((x) => x.json()),
        }),
      },
    ],
  },
];
