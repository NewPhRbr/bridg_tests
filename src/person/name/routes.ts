import * as api from "./api";
import Edit from "./edit";

export default [
  {
    path: ":nameId",
    children: [
      {
        path: "edit",
        Component: Edit,
        loader: ({ params: { personId, nameId } }) => api.find(personId, nameId),
        handle: {
          breadcrumb: "Rename",
        },
      },
    ],
  },
];
