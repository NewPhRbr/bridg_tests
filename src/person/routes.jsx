import * as api from "./api";
import Index from "./index";
import Edit from "./edit";

export default [
  {
    path: "",
    element: <Index />,
  },
  {
    path: ":id",
    loader: ({ params: { id } }) => api.find(id),
    Component: Edit,
  },
];
