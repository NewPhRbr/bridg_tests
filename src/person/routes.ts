import * as api from "./api";
import Edit from "./edit";
import Index from "./index";
import nameRoutes from "./name/routes";
import New from "./new";
import Show from "./show";

export default [
  {
    handle: {
      breadcrumb: "Persons",
    },
    children: [
      {
        path: "",
        Component: Index,
      },
      {
        path: "new",
        Component: New,
        handle: {
          breadcrumb: "New",
        },
      },
      {
        id: "person",
        path: ":personId",
        loader: ({ params: { personId } }: any) => api.find(personId),
        handle: {
          breadcrumb: (data) => data.name?.full,
        },
        children: [
          {
            path: "",
            Component: Show,
            loader: ({ params: { personId } }: any) => api.find(personId),
          },
          {
            path: "edit",
            Component: Edit,
            loader: ({ params: { personId } }: any) => api.find(personId),
            handle: {
              breadcrumb: "Edit",
            },
          },
          {
            path: "name",
            children: nameRoutes,
          },
        ],
      },
    ],
  },
];
