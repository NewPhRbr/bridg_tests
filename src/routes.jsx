import Layout from "./Layout";
import personRoutes from "./person/routes";

export default [
  {
    path: "/",
    element: <Layout />,
    children: [
      {
        path: "",
        element: <div>Works</div>,
      },
      {
        path: "/persons",
        children: personRoutes,
      },
    ],
  },
];
