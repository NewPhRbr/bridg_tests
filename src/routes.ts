import Layout from "./Layout";
import personRoutes from "./person/routes";

export default [
  {
    path: "/",
    Component: Layout,
    children: [
      {
        path: "",
      },
      {
        path: "/persons",
        children: personRoutes,
      },
    ],
  },
];
