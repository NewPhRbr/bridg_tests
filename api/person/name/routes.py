from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import orm

from api.db import get_db
from api.person.name import schema
from api.util import setattrs
from umdb import common as model

router = APIRouter(prefix="/persons/{person_id}/names", tags=["names"])


@router.get("/", response_model=list[schema.Name])
def index(person_id: int, db: orm.Session = Depends(get_db)):
    return db.query(model.Name).where(model.Name.person_id == person_id).all()


@router.get("/{id}", response_model=schema.Name)
def find(person_id: int, id: int, db: orm.Session = Depends(get_db)):
    name = db.query(model.Name).where(model.Name.id == id).first()
    if name is None:
        raise HTTPException(404)
    return name


@router.patch("/{id}", response_model=schema.Name)
def update(
    person_id: int,
    id: int,
    body: schema.NameUpdate,
    db: orm.Session = Depends(get_db),
):
    name = db.query(model.Name).where(model.Name.id == id).first()
    if name is None:
        raise HTTPException(404)
    name = setattrs(name, body.model_dump())
    db.add(name)
    db.commit()
    db.refresh(name)
    return name
