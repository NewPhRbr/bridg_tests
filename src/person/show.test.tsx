import { render, screen } from "@testing-library/react";
import { HttpResponse, http } from "msw";
import { setupServer } from "msw/node";
import { RouterProvider, createMemoryRouter } from "react-router-dom";
import { afterAll, afterEach, beforeAll, expect, test } from "vitest";
import routes from "../routes";

const DATA = {
  sex: "M",
  birth_date: "2024-05-21",
  death_date: "2024-05-21",
  death_date_estimated_indicator: true,
  death_indicator: true,
  id: 1,
  primary_name: {
    use: "string",
    family: "John",
    given: "Doe",
    middle: "",
    patronymic: "",
    prefix: "",
    suffix: "",
    id: 2,
    full: "Doe John",
  },
};

const server = setupServer(http.get("/api/persons/1", () => HttpResponse.json(DATA)));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("page renders correctly", async () => {
  const router = createMemoryRouter(routes, { initialEntries: ["/persons/1"] });
  const tree = render(<RouterProvider router={router} />);
  await screen.findAllByText("Doe John");
  expect(tree).toMatchSnapshot();
});
