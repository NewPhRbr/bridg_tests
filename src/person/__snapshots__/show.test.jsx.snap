// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`page renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="mb-1 fs-7"
          style="--bs-breadcrumb-divider: '>';"
        >
          <ol
            class="breadcrumb my-auto"
          >
            <li
              class="breadcrumb-item  undefined"
            >
              <a
                href="/"
              >
                Persons
              </a>
            </li>
            <li
              class="breadcrumb-item active undefined"
            >
              Doe John
            </li>
          </ol>
        </nav>
        <div
          class="d-flex undefined"
        >
          <div
            class="flex-grow-1"
          >
            <div
              class="lead"
            >
              <span
                class="fs-4 me-2"
              >
                Doe John
              </span>
              <span
                class="fs-6"
              >
                2024-05-21
              </span>
            </div>
          </div>
          <div
            class="ps-4"
          >
            <div
              class="dropdown"
            >
              <button
                class="btn btn-primary btn-sm dropdown-toggle"
                data-bs-toggle="dropdown"
                type="button"
              >
                Actions
              </button>
              <ul
                class="dropdown-menu"
              >
                <li>
                  <a
                    class="dropdown-item"
                    href="#"
                  >
                    Rename
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="row my-3"
        >
          <div
            class="col-sm-4"
          >
            <div
              class="card undefined"
            >
              <div
                class="card-header bg-body"
              >
                <div
                  class="d-flex"
                >
                  <div
                    class="lead fs-5"
                  >
                    Information
                  </div>
                  <div
                    class="ms-auto"
                  >
                    <a
                      class="fs-8 align-middle"
                      href="/edit"
                    >
                      <i
                        class="fa-solid fa-pencil pe-1"
                      />
                       Edit
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="card-body undefined"
              >
                <pre
                  class="mb-0"
                >
                  {
  "sex": "M",
  "birth_date": "2024-05-21",
  "death_date": "2024-05-21",
  "death_date_estimated_indicator": true,
  "death_indicator": true,
  "id": 1
}
                </pre>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="mb-1 fs-7"
        style="--bs-breadcrumb-divider: '>';"
      >
        <ol
          class="breadcrumb my-auto"
        >
          <li
            class="breadcrumb-item  undefined"
          >
            <a
              href="/"
            >
              Persons
            </a>
          </li>
          <li
            class="breadcrumb-item active undefined"
          >
            Doe John
          </li>
        </ol>
      </nav>
      <div
        class="d-flex undefined"
      >
        <div
          class="flex-grow-1"
        >
          <div
            class="lead"
          >
            <span
              class="fs-4 me-2"
            >
              Doe John
            </span>
            <span
              class="fs-6"
            >
              2024-05-21
            </span>
          </div>
        </div>
        <div
          class="ps-4"
        >
          <div
            class="dropdown"
          >
            <button
              class="btn btn-primary btn-sm dropdown-toggle"
              data-bs-toggle="dropdown"
              type="button"
            >
              Actions
            </button>
            <ul
              class="dropdown-menu"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Rename
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="row my-3"
      >
        <div
          class="col-sm-4"
        >
          <div
            class="card undefined"
          >
            <div
              class="card-header bg-body"
            >
              <div
                class="d-flex"
              >
                <div
                  class="lead fs-5"
                >
                  Information
                </div>
                <div
                  class="ms-auto"
                >
                  <a
                    class="fs-8 align-middle"
                    href="/edit"
                  >
                    <i
                      class="fa-solid fa-pencil pe-1"
                    />
                     Edit
                  </a>
                </div>
              </div>
            </div>
            <div
              class="card-body undefined"
            >
              <pre
                class="mb-0"
              >
                {
  "sex": "M",
  "birth_date": "2024-05-21",
  "death_date": "2024-05-21",
  "death_date_estimated_indicator": true,
  "death_indicator": true,
  "id": 1
}
              </pre>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
